lib/polariscope/scanner/calculation_context.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Polariscope
+  module Scanner
+    class CalculationContext
+      DEPENDENCY_PRIORITIES = { rails: 10.0 }.freeze
+      GROUP_PRIORITIES = { default: 2.0, production: 2.0 }.freeze
+      DEFAULT_DEPENDENCY_PRIORITY = 1.0
+
+      ADVISORY_SEVERITY = 1.09
+      ADVISORY_PENALTIES = {
+        none: 0.0,
+        low: 0.5,
+        medium: 1.0,
+        high: 3.0,
+        critical: 5.0
+      }.freeze
+      FALLBACK_ADVISORY_PENALTY = 0.5
+
+      MAJOR_VERSION_PENALTY = 1
+      NEW_VERSIONS_SEVERITY = 1.07
+      SEGMENT_SEVERITIES = [1.7, 1.15, 1.01].freeze
```

Comment:
Previously, this constant had values `[1.7, 1.15, 1.01, 1.005]`. In practice, the last segment isn't very useful. For example, comparing versions v7.0.0 and v7.0.0.1 will not cause a health score difference because the former version only has three segments, so comparison only makes sense if you have versions like v7.0.0.1 and v7.0.0.2.

Because of that, and the fact that Semver doesn't define a fourth segment and its meaning is defined by dependency owners (like Rails, which uses it for security releases), I've decided to remove it. In the case of Rails, health score will still be impacted by advisories because v7.0.0.1 will have an advisory that v7.0.0.2 won't.